File: SSVMpack/SSVMpack/demonSyntData.m
% demon randomly generated data
clc; close all; clear all;  addpath(genpath(pwd));
 
type    = 1;   % 1 or 2 or 3
Ex      = {'2D', '3D', 'nD'};
m0      = 4e2;  
n0      = 100;
[Atrain,ytrain,Atest,ytest] = randomData(Ex{type},m0,n0,0);  
[m, n]  = size(Atrain); 

t       = 1;
solver  = {'NM01','NSSVM'};
pars.C  = 0.25;
out     = SSVMpack(Atrain,ytrain,solver{t},pars);  
acc     = accuracy(Atrain,out.w,ytrain); 
tacc    = accuracy(Atest,out.w,ytest);
fprintf(' Training  Time:             %5.3fsec\n',out.time);
fprintf(' Training  Size:             %dx%d\n',m,n);
fprintf(' Training  Accuracy:         %5.2f%%\n', acc*100) 
fprintf(' Testing   Size:             %dx%d\n',size(Atest,1),n);
fprintf(' Testing   Accuracy:         %5.2f%%\n',tacc*100);
fprintf(' Number of Support Vectors:  %d\n',out.sv); 
if isequal(Ex{type},'2D') && m <400
   figure('Renderer', 'painters', 'Position', [900, 400, 300 300])
   plot2D(Atrain,ytrain,out.w,solver{t},acc); xlabel('Training data')
   figure('Renderer', 'painters', 'Position', [1220, 400, 300 300])
   plot2D(Atest,ytest,out.w,solver{t},tacc); xlabel('Testing data')
end

File: SSVMpack/SSVMpack/demonSyntData.py
import numpy as np

try:
    from .SSVMpack import SSVMpack as run_ssvmpack
    from .SSVMpack.funcs.randomData import randomData
    from .SSVMpack.funcs.accuracy import accuracy
    from .SSVMpack.funcs.plot2D import plot2D
except Exception:  # pragma: no cover
    import os
    import sys

    repo_root = os.path.dirname(os.path.dirname(__file__))
    sys.path.insert(0, repo_root)
    from SSVMpack.SSVMpack import SSVMpack as run_ssvmpack
    from SSVMpack.SSVMpack.funcs.randomData import randomData
    from SSVMpack.SSVMpack.funcs.accuracy import accuracy
    from SSVMpack.SSVMpack.funcs.plot2D import plot2D


def main():
    kind_idx = 0  # 0 -> '2D', 1 -> '3D', 2 -> 'nD'
    kinds = ['2D', '3D', 'nD']
    m0 = int(4e2)
    n0 = 100
    Atrain, ytrain, Atest, ytest = randomData(kinds[kind_idx], m0, n0, 0.0)
    m, n = Atrain.shape

    solver = ['NM01', 'NSSVM']
    pars = {'C': 0.25}
    out = run_ssvmpack(Atrain, ytrain, solver[0], pars)

    acc, _, _ = accuracy(Atrain, out['x'], ytrain)
    tacc, _, _ = accuracy(Atest, out['x'], ytest)

    print(f" Training  Time:             {out['time']:.3f}sec")
    print(f" Training  Size:             {m}x{n}")
    print(f" Training  Accuracy:         {acc * 100:5.2f}%")
    print(f" Testing   Size:             {Atest.shape[0]}x{n}")
    print(f" Testing   Accuracy:         {tacc * 100:5.2f}%")
    print(f" Number of Support Vectors:  {out['sv']}")

    if kinds[kind_idx] == '2D' and m < 400:
        plot2D(Atrain, ytrain, out['x'], solver[0], acc)
        plot2D(Atest, ytest, out['x'], solver[0], tacc)


if __name__ == '__main__':
    main()
